src/server/worker/api_server.go

```diff
@@ -976,6 +976,15 @@ func (a *APIServer) uploadOutput(pachClient *client.APIClient, dir string, tag s
 	if err := putObjsClient.CloseSend(); err != nil {
 		return err
 	}
+	// Be sure that the stream finished successfully.
+	for {
+		if err := putObjsClient.RecvMsg(nil); err != nil {
```

Comment:
Won't this panic if you don't pass in a message to deserialize into?

Reply:
No, it does not, but after thinking about it a bit, I think it would be best to not rely on that behavior. The order of operations for grpc are: 
```
read the next message:
  if no messages are read and the stream has closed successfully return io.EOF 
  else deserialize the message
```
I just noticed while looking at the grpc code that I actually have a CloseAndRecv function available to do both of these operations, so I am just going to switch to that.